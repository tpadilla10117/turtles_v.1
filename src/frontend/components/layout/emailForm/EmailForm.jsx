import React from 'react';
import {
  CtaButton
} from '../../utils';

/* TODO: Finish styling */
function EmailForm( {bannerImg, alt} ) {
  return (
    <div className='emailForm-parent-container'>
      <img src={bannerImg} alt={alt} className='emailForm-img' />

      <div className='emailForm-header-parent-container'>
        <h1 className='emailForm-header'>Contact Me</h1>
      </div>

      <div className='emailForm-wrapper'>
        <form className='emailForm-form'>
          <div className='emailForm-form-input-wrapper'>
            <input id='name-input' className='form-input' type='text' placeholder='Name' name='user_name' required />
            <label htmlFor='name-input'></label>
            <input id='subject-input' className='form-input' type='text' placeholder='Subject' name='user_subject' required />
            <label htmlFor='subject-input'></label>
            <input id='email-input' className='form-input' type='text' placeholder='Email' name='user_email' required />
            <label htmlFor='email-input'></label>
            <textarea id='message-input' className='form-input' rows='5' placeholder='Message' name='message'/>
            <label htmlFor='message-input'></label>
          </div>

          <CtaButton 
            classnames={'emailForm-form-submitBtn'}
            text={'Submit'}
          />

        </form>

      </div>

    </div>
  )
}

export default EmailForm;