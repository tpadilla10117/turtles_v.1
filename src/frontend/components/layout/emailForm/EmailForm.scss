.emailForm-parent-container {
    width: 100%;
    height: 100%;
    overflow: hidden;
    position: relative;
    background: hsla(0,0%,42.7%,.12);
    box-shadow: 0 3px 30px rgb(0 0 0 / 10%);
    backdrop-filter: blur(5px);


    &:before {
        content: "";
        position: absolute;
        width: 100%;
        height: 100%;
        z-index: 1;
        background: linear-gradient(to top, $background-complement 25%, transparent);
    }

    .emailForm-img {
        position: absolute;
        object-fit: cover;
        height: 600px;
        top: -220px;
    }

    .emailForm-header-parent-container {
        position: absolute;
        z-index: 2;
        top: 15%;

        .emailForm-header {
            color: $text-secondary;
            font-family: $header-primaryfont;
            font-weight: 400;
            font-size: $heading-primary;
            /* TODO: Needs shadow */
            text-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
        }
    }

    .emailForm-wrapper {
        position: absolute;
        z-index: 2;
        width: 100%;
        height: 80%;
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        padding-top: 150px;

        .emailForm-form {
            width: 260px;
            display: flex;
            flex-direction: column;
            justify-content: space-evenly;
            align-items: center;

            input {
                width: 330px;
                height: 30px;
                outline: none;
                border: none;
                background: $text-secondary;
                border-radius: 5px;
                padding: 0 10px;
                margin: 10px 0px;
                font-family: $button-primaryfont;
                font-size: 14px; /* TODO: Change */
                
                &::placeholder {
                    font-family: $button-primaryfont;
                } 
            }

            textarea {
                width: 330px;
                height: 120px;
                outline: none;
                border: none;
                border-radius: 5px;
                background: $text-secondary;
                padding: 0 10px;
                padding-top: 10px;
                margin: 10px 0px;
                font-family: $button-primaryfont;
                font-size: 14px; /* TODO: Change */
                
                &::placeholder {
                    font-family: $button-primaryfont;
                } 
                
            }
        }

    }
}