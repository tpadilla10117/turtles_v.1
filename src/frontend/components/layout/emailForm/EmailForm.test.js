import { shallow } from 'enzyme';
import React from 'react';
import EmailForm from './EmailForm';

describe('The EmailForm.jsx component:', () => {
    let wrapper;

    beforeEach( () => {
        const mockProps = {
            bannerImg: '',
            alt: ''
        };

        wrapper = shallow(<EmailForm {...mockProps} alt='text' bannerImg='text.jpeg'/>)
        
    });

    it('renders the EmailForm.jsx component', () => {
        expect.assertions(2);

        expect(wrapper.length).toEqual(1);
        expect(wrapper).toMatchSnapshot();
    });

    /* it('renders out the correct props', () => {
        const mockProps = {
            bannerImg: '',
            alt: 'text'
        };
        
        let wrapper = shallow(<EmailForm alt={mockProps.alt} />)
        expect(wrapper.prop(mockProps.alt)).toEqual('text');

        
    }) */
})