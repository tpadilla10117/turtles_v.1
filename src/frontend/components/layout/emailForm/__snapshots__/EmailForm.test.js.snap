// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The EmailForm.jsx component: renders the EmailForm.jsx component 1`] = `ShallowWrapper {}`;
